repositories/dbmodels/db_decision_rule.go

```diff
@@ -1,44 +1,90 @@
 package dbmodels
 
 import (
+	"errors"
 	"fmt"
 
 	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/models/ast"
 	"github.com/checkmarble/marble-backend/utils"
 
 	"github.com/jackc/pgx/v5/pgtype"
 )
 
 type DbDecisionRule struct {
-	Id             string      `db:"id"`
-	OrganizationId string      `db:"org_id"`
-	DecisionId     string      `db:"decision_id"`
-	Name           string      `db:"name"`
-	Description    string      `db:"description"`
-	ScoreModifier  int         `db:"score_modifier"`
-	Result         bool        `db:"result"`
-	ErrorCode      int         `db:"error_code"`
-	DeletedAt      pgtype.Time `db:"deleted_at"`
-	RuleId         string      `db:"rule_id"`
-	RuleEvaluation []byte      `db:"rule_evaluation"`
+	Id             string         `db:"id"`
+	OrganizationId string         `db:"org_id"`
+	DecisionId     string         `db:"decision_id"`
+	Name           string         `db:"name"`
+	Description    string         `db:"description"`
+	ScoreModifier  int            `db:"score_modifier"`
+	Result         bool           `db:"result"`
+	ErrorCode      ExecutionError `db:"error_code"`
+	DeletedAt      pgtype.Time    `db:"deleted_at"`
+	RuleId         string         `db:"rule_id"`
+	RuleEvaluation []byte         `db:"rule_evaluation"`
 }
 
 const TABLE_DECISION_RULES = "decision_rules"
 
 var SelectDecisionRuleColumn = utils.ColumnList[DbDecisionRule]()
 
-func adaptErrorCodeAsError(errCode models.ExecutionError) error {
+type ExecutionError int
+
+const (
+	NoError              ExecutionError = 0
+	DivisionByZero       ExecutionError = 100
+	NullFieldRead        ExecutionError = 200
+	NoRowsRead           ExecutionError = 201
+	PayloadFieldNotFound ExecutionError = 202
+	Unknown              ExecutionError = -1
+)
```

Comment:
Some context on the current code :
- this error is a "single error" attached to each rule evaluation
- the error is derived from the flatten rule evaluation errors (the logic which choose the single error is defined in `AdaptExecutionError` above)
- errors codes are only used for storage ATM (artificially rebuild the error on READ using `adaptErrorCodeAsError`)

This could be handled totally differently :
1. **remove**: we do store every nested errors in the rule execution now, so this "root" error is a sugar prop
2. **kept**: we could adapt `error` to `EvaluationErrorDto` as we do for other evaluation errors

Why I did not choose one of the above solutions : 
- this error "root" prop is documented in [our public API](https://docs.checkmarble.com/reference/ressource-definitions#error):
  - in practice, we always return `code: 1` so error codes are not really respecting the doc
  - explanation of error code `100` looks wrong (comparing public API vs our codebase)
- on existing decisions, this is the only valuable stored information on decision Errors

What we could do :
- Choose the 2. solution (since we do not respect public API error codes, it's not really a problem to break IMO)
- Migrate existing decisions to store equivalent `EvaluationErrorDto` (so we do not loose information on past decisions)

Reply:
> in practice, we always return code: 1 so error codes are not really respecting the doc

I don't think that's the case, the error code is properly stored on rule executions (however there is an error code on the decision level which may be always at 1)

Reply:
> explanation of error code 100 looks wrong (comparing public API vs our codebase)

is it ? 🤔

Reply:
Taken from the documentation :
code | message
-- | --
100 | Scenario was not able to compute a score because all rules failed.

What is returned for code 100 in our code base :
code | message
-- | --
1 | Division by zero (alike message)

-> All rules did not failed, the error is specific to Division by zero